### PR TITLE
feat: Remove staking API, extract @0x/api-utils

### PR DESCRIPTION
**Pull request by xianny**
# Description

* Replace a bunch of generics with [@0x/api-utils](https://github.com/0xProject/api-utils)
* Remove staking API

Moved staking to https://github.com/0xProject/0x-event-pipeline/pull/52

Next Steps:

- [x] Finish  https://github.com/0xProject/0x-event-pipeline/pull/52
- [x] Update 0x-main-infra to direct `/staking` ingress to an independent staking-api image, instead of the main 0x-api image https://github.com/0xProject/0x-main-infra/pull/1211
- [x] Test and make sure everything works
- [ ] Merge this and use as the 0x-api image everywhere

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.
-   [ ] Test changes on the staging environment with [Matcha API Staging](https://api-staging.matcha.xyz).

    -   [ ] SRA/Limit orders
    -   [ ] Swap endpoints
    -   [ ] Meta transaction endpoints
    -   [ ] Depth charts

    For more information see `0x API Matcha smoke test runbook` in Quip.

### CODE REVIEW

File: src/runners/http_sra_service_runner.ts [hunk 1]
```diff
@@ -1,25 +1,25 @@
 /**
  * This module can be used to run the SRA HTTP service standalone
  */
+import { cacheControl, createDefaultServer } from '@0x/api-utils';
 import * as express from 'express';
 // tslint:disable-next-line:no-implicit-dependencies
 import * as core from 'express-serve-static-core';
```
Comment (kimpers) [inline src/runners/http_sra_service_runner.ts:7]: NIT: would be super nice to move all these base server modules into `@0x/api-utils` and then just export them from the package. We can then easily bump versions in `@0x/api-utils` and all services will automatically get the new versions when they update `@0x/api-utils` ✨ 🧙

File: src/runners/http_staking_service_runner.ts [hunk 2]
```diff
@@ -1,59 +0,0 @@
-/**
```
Comment (kimpers) [inline src/runners/http_staking_service_runner.ts:1]: Is staking finally getting its own service? 🙏  Let's make sure that the website is correctly hooked up before we ship this 😄

**Review by kimpers — APPROVED** [review]
AWESOME! Thanks for doing this! This will make spinning up and maintaining our services so much easier 🙏 🙏

File: src/runners/http_sra_service_runner.ts (hunk 1 above)
Comment (xianny) [reply src/runners/http_sra_service_runner.ts:7]: Love this idea!

**Comment by xianny** [comment]
deployed on staging, Assertible is green 🤷

**Pull request merged**